src/garage/torch/modules/multi_headed_mlp_module.py

```diff
@@ -61,55 +64,43 @@ def __init__(self,
 
         output_dims = self._check_hidden_layer_parameter(
             'output_dims', output_dims, n_heads)
-        output_nonlinearities = self._check_hidden_layer_parameter(
-            'output_nonlinearities', output_nonlinearities, n_heads)
         output_w_inits = self._check_hidden_layer_parameter(
             'output_w_inits', output_w_inits, n_heads)
         output_b_inits = self._check_hidden_layer_parameter(
             'output_b_inits', output_b_inits, n_heads)
 
-        prev_size = input_dim
-        for size in hidden_sizes:
-            layer = nn.Sequential()
-            if layer_normalization:
-                layer.add_module('layer_normalization',
-                                 nn.LayerNorm(prev_size))
-
-            linear_layer = nn.Linear(prev_size, size)
-            hidden_w_init(linear_layer.weight)
-            hidden_b_init(linear_layer.bias)
-            layer.add_module('linear', linear_layer)
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._layer_normalization = layer_normalization
+        self._output_nonlinearities = self._check_hidden_layer_parameter(
+            'output_nonlinearities', output_nonlinearities, n_heads)
 
-            if hidden_nonlinearity:
-                layer.add_module('non_linearity', hidden_nonlinearity)
+        self._layers = nn.ModuleList()
 
+        prev_size = input_dim
+        for size in hidden_sizes:
+            layer = nn.Linear(prev_size, size)
+            hidden_w_init(layer.weight)
+            hidden_b_init(layer.bias)
```

Comment:
@krzentner do we want to potentially setup the nn.ModuleList of each PyTorch module to contain the nonlinearities inside the module list. I think currently, they are explicitly run inside the forward function of each module, but they aren't explicitly apart of the modulelist (by using torch.functional nonlinearities instead of using nn.module nonlinearities)

Reply:
Can we add the `torch.nn.functional` nonlinearities to the module list? That seems like a safer option, in case someone provides a `nn.Module` based nonlinearity that has parameters.

It would be pretty bad if the user can pass in a nonlinearity with parameters and we just forget to train those parameters.

Reply:
To be clear: we will only be using torch.nn.nonlinearities in our modules, because we want them to be apart of the module list? I don't think one can add torch.functional non linearities to an nn.ModuleList @krzentner